text/0083-sbt-standard.md

```diff
@@ -0,0 +1,252 @@
+
+**Should be rejected if:**
+1. `signature` verification failed.
+2. `sbt_nonce` not equals stored nonce.
+3. `new_owner` is not sender's address.
+4. there is not enough coins (with respect to NFT own storage fee guidelines).
+5. After processing the request, the contract must send at least in_msg_value - forward_amount - max_tx_gas_price to the response_destination address. If the contract cannot guarantee this, it must immediately stop executing the request and throw the error.
+
+**Otherwise should do:**
+1. Change current owner of SBT to `new_owner` address.
+2. Generate and store new random `nonce`.
+3. Send all excesses of incoming message coins to response_destination with the following layout:
+   
+   `excesses#d53276db query_id:uint64 = InternalMsgBody;`
+
+   query_id should be equal with request's `query_id`.
+
+#### 2. `prove_ownership`
+
+TL-B schema of inbound message:
+```
+prove_ownership#5c9b0fb1 query_id:uint64 dest:MsgAddress 
+data:^Cell with_content:bool = InternalMsgBody;
+```
+`query_id` -  arbitrary request number.
+
+`dest` -  address of the contract to which the ownership of SBT should be proven.
+
+`data` - arbitrary data required by target contract.
+
+`with_content` - if true, SBT's content cell will be included in message to contract.
+
+**Should do:**
+
+Send message with TL-B schema to `dest` contract:
+```
+verify_ownership#a553079c query_id:uint64 sbt_id:uint256 owner:MsgAddress 
+data:^Cell content:(Maybe ^Cell) = InternalMsgBody;
+```
+
+`query_id` - request number passed in `prove_ownership`.
+
+`sbt_id` -  id of SBT.
+
+`owner` - current owner's address.
+
+`data` - data cell passed in `prove_ownership`.
+
+`content` - SBT's content, it is passed if `with_content` was true in `prove_ownership`.
+
+In case when `verify_ownership` was bounced back to SBT, SBT should send message to owner with schema:
+```
+verify_ownership_bounced#81b510c2 query_id:uint64 sbt_id:uint256 owner:MsgAddress 
+data:^Cell content:(Maybe ^Cell) = InternalMsgBody;
+```
+
+#### 3. `destroy`
+
+TL-B schema of inbound message:
+```
+destroy#1f04537a query_id:uint64 = InternalMsgBody;
+```
+`query_id` -  arbitrary request number.
+
+**Should be rejected if:**
+Sender address is not an owner's address.
+
+**Otherwise should do:**
+Set owner's address to null and set public key to 0.
+
+#### 4. `revoke`
+
+TL-B schema of inbound message:
+```
+revoke#6f89f5e3 query_id:uint64 = InternalMsgBody;
+```
+`query_id` -  arbitrary request number.
+
+**Should be rejected if:**
+Sender address is not an authority's address.
+
+**Otherwise should do:**
+Set owner's address to null and set public key to 0.
+
+**GET methods**
+1. `get_public_key()` - returns `int`, that is owner's public key.
+2. `get_nonce()` - returns `int`, which current nonce.
+3. `get_nft_data()` - same as in [NFT standard](https://github.com/ton-blockchain/TIPs/issues/62).
+4. `get_authority_address()` - returns `slice`, that is authority's address. Authority can revoke SBT.
```

Comment:
If authority can not be changed may be it is better to force collection to be authority? Not sure though

Reply:
Then it will require custom collection to work with SBT, with additional revoke method, not sure if it will be good.